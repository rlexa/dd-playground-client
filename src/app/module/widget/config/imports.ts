import { CommonModule } from '@angular/common';
import { MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';

export const imports = [CommonModule, RouterModule, MatCardModule, MatListModule];
