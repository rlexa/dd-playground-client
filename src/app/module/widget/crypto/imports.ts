import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import {
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
} from '@angular/material';
import { RouterModule } from '@angular/router';
import { CryptoApiModule } from 'app/module/service/crypto-api';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'app/module/directive/flexbox';

export const imports = [
  CommonModule,
  FormsModule,
  RouterModule,
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
  FlexboxModule,
  CryptoApiModule,
  SimpleViewModule,
];
