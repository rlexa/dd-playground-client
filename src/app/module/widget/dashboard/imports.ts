import { CommonModule } from '@angular/common';
import { MatButtonModule, MatToolbarModule, MatTooltipModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { IconPipeModule } from 'app/module/pipe/icon';
import { RoutedContentModule } from 'app/module/widget/routed-content';
import { VersionModule } from 'app/module/widget/version';
import { WidgetsModule } from 'app/widgets/widgets.module';
import { FlexboxModule } from 'dd-flexbox';
import 'hammerjs';

export const imports = [
  CommonModule, RouterModule,
  MatButtonModule, MatToolbarModule, MatTooltipModule, FlexboxModule,
  IconPipeModule, VersionModule,
  // remove
  RoutedContentModule, WidgetsModule,
];
