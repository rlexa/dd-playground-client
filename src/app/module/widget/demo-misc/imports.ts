import { CommonModule } from '@angular/common';
import { MatButtonModule, MatCardModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { GraphskyApiModule } from 'app/module/service/graphsky-api';
import { DiagramPolynomModule } from 'app/module/widget/diagram-polynom';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'app/module/directive/flexbox';

export const imports = [
  CommonModule,
  RouterModule,
  MatButtonModule,
  MatCardModule,
  FlexboxModule,
  DiagramPolynomModule,
  GraphskyApiModule,
  SimpleViewModule,
];
