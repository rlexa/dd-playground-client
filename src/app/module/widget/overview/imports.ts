import { CommonModule } from '@angular/common';
import { MatButtonModule, MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { RoutedContentModule } from 'app/module/widget/routed-content';
import { FlexboxModule } from 'dd-flexbox';

export const imports = [CommonModule, RouterModule, MatButtonModule, MatCardModule, MatListModule, FlexboxModule, RoutedContentModule];
