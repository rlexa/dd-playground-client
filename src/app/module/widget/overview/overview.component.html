<mat-card>
  <mat-card-title>Empty?</mat-card-title>
  <mat-card-content>
    <mat-list>
      <h4 mat-header>Empty</h4>
    </mat-list>
  </mat-card-content>
</mat-card>
