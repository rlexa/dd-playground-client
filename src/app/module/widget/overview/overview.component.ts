import { ChangeDetectionStrategy, Component } from '@angular/core';
import { Router } from '@angular/router';
import { routeToAiApproxPolynom } from 'app/routing';

@Component({
  selector: 'app-overview',
  templateUrl: './overview.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class OverviewComponent {
  constructor(private readonly router: Router) { }
  gotoAiApproxPolynom = () => routeToAiApproxPolynom(this.router);
}
