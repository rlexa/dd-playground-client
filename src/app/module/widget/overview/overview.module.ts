import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { ROUTE_ROOT, ROUTE_WILDCARD } from 'app/routing';
import { imports } from './imports';
import { OverviewComponent } from './overview.component';

const ROUTING = [
  { path: ROUTE_ROOT, component: OverviewComponent },
  { path: ROUTE_WILDCARD, redirectTo: ROUTE_ROOT, pathMatch: 'full' },
];

@NgModule({ imports: [...imports, RouterModule.forChild(ROUTING)], exports: [OverviewComponent], declarations: [OverviewComponent] })
export class OverviewModule { }
