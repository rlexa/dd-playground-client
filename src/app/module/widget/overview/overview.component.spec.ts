import { async, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { imports } from './imports';
import { OverviewComponent } from './overview.component';

describe('OverviewComponent', () => {
  beforeEach(async(() => {
    TestBed.configureCompiler(<any>{ preserveWhitespaces: false }).configureTestingModule({
      imports: [NoopAnimationsModule, RouterTestingModule, ...imports], declarations: [OverviewComponent], providers: [],
    }).compileComponents();
  }));

  test('is created', () => {
    const fixture = TestBed.createComponent(OverviewComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
