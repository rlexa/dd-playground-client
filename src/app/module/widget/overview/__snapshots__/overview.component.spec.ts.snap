// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewComponent is created 1`] = `
<app-overview
  gotoAiApproxPolynom={[Function Function]}
  router={[Function Router]}
>
  <div
    ddflexbox="hyscsw"
    ng-reflect-dd-flexbox="hyscsw"
    style="display: flex; flex-direction: row; justify-content: flex-start; align-items: flex-start; align-content: flex-start; flex-wrap: wrap;"
  >
    <mat-card
      class="mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        ML (Machine Learning)
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <mat-list
          class="mat-list mat-list-base"
        >
          <h4
            mat-header=""
          >
            Tensorflow
          </h4>
          <button
            class="mat-raised-button _mat-animation-noopable"
            mat-raised-button=""
          >
            <span
              class="mat-button-wrapper"
            >
              Approximate Polynomial
            </span>
            <div
              class="mat-button-ripple mat-ripple"
              matripple=""
              ng-reflect-centered="false"
              ng-reflect-disabled="false"
              ng-reflect-trigger="[object HTMLButtonElement]"
            />
            <div
              class="mat-button-focus-overlay"
            />
          </button>
        </mat-list>
      </mat-card-content>
    </mat-card>
  </div>
</app-overview>
`;
